usage/rest-server/src/main/java/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -216,9 +222,18 @@ public JsonNode fetch(String entityIds) {
 
     @Override
     public List<ApplicationSummary> list() {
+        return list(".*");
+    }
+
+    @Override
+    public List<ApplicationSummary> list(String typeRegex) {
+        if (Strings.isBlank(typeRegex)) {
+            typeRegex = ".*";
+        }
         return FluentIterable
                 .from(mgmt().getApplications())
                 .filter(EntitlementPredicates.isEntitled(mgmt().getEntitlementManager(), Entitlements.SEE_ENTITY))
+                .filter(EntityPredicates.hasInterfaceMatching(typeRegex))
```

Comment:
For the record, another option would be to use `EntityType.getName()`, but I prefer yours - good to be able to find everything of type for example `DynamicCluster` including sub-types.